src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs

```diff
@@ -38,41 +117,84 @@ public static bool GetShowSelected(DependencyObject element)
 
         private static void ShowSelectedPropertyChanged(DependencyObject obj, DependencyPropertyChangedEventArgs args)
```

Comment:
I got you now, thank you for these comments helping me to understand. The implementation here does seem breakable. Maybe we can create several function as to color mesh/point&line in different condition-> isolate, frozen, selection because it seems we always need to go through these cases again and again.